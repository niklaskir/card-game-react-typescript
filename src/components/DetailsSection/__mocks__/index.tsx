import React from 'react';

const DetailsSection = () => <div id="details_section" />;

export default DetailsSection;
