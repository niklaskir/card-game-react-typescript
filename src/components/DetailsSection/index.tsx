import React from 'react';
import { useRecoilValue } from 'recoil';
import { userSelected as userSelectedState } from '../../recoil/states/userList';

const OverviewSection = () => {
  const selectedUser = useRecoilValue(userSelectedState);

  return (
    <div className="col-span-1 lg:col-span-2">
      <div className="rounded overflow-hidden shadow-lg bg-slate-100">
        <div className="px-6 py-4">
          <div className="font-bold text-xl mb-2">Details</div>
          {selectedUser ? (
            <ul className="list-disc px-6">
              <li className="text-gray-700 text-xl">{selectedUser.realName}</li>
              <li className="text-gray-700 text-xl">
                {selectedUser.playerName}
              </li>
              <li className="text-gray-700 text-xl">{selectedUser.asset}</li>
            </ul>
          ) : (
            <div className="text-xl mb-2">Please select user...</div>
          )}
        </div>
      </div>
    </div>
  );
};

export default OverviewSection;
