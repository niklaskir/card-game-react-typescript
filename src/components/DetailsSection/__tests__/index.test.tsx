import React from 'react';
import Recoil from 'recoil';
import renderer from 'react-test-renderer';
import DetailsSection from '../index';

jest.mock('recoil', () => ({
  atom: jest.fn(),
  selector: jest.fn(),
  useRecoilValue: jest.fn(),
}));

describe('DetailsSection', function () {
  it('should render correctly with selectedUser', () => {
    Recoil.useRecoilValue = jest.fn().mockReturnValueOnce({
      realName: 'Sample real name',
      playerName: 'Sample player name',
      asset: 'Sample asset',
    });

    const detailsSection = renderer.create(<DetailsSection />).toJSON();

    expect(detailsSection).toMatchSnapshot();
  });

  it('should render correctly when selectedUser is null', () => {
    const detailsSection = renderer.create(<DetailsSection />).toJSON();

    expect(detailsSection).toMatchSnapshot();
  });
});
