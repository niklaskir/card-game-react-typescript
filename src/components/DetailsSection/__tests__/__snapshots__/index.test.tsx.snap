// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsSection should render correctly when selectedUser is null 1`] = `
<div
  className="col-span-1 lg:col-span-2"
>
  <div
    className="rounded overflow-hidden shadow-lg bg-slate-100"
  >
    <div
      className="px-6 py-4"
    >
      <div
        className="font-bold text-xl mb-2"
      >
        Details
      </div>
      <div
        className="text-xl mb-2"
      >
        Please select user...
      </div>
    </div>
  </div>
</div>
`;

exports[`DetailsSection should render correctly with selectedUser 1`] = `
<div
  className="col-span-1 lg:col-span-2"
>
  <div
    className="rounded overflow-hidden shadow-lg bg-slate-100"
  >
    <div
      className="px-6 py-4"
    >
      <div
        className="font-bold text-xl mb-2"
      >
        Details
      </div>
      <ul
        className="list-disc px-6"
      >
        <li
          className="text-gray-700 text-xl"
        >
          Sample real name
        </li>
        <li
          className="text-gray-700 text-xl"
        >
          Sample player name
        </li>
        <li
          className="text-gray-700 text-xl"
        >
          Sample asset
        </li>
      </ul>
    </div>
  </div>
</div>
`;
